hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1453,6 +1453,15 @@ public Optional<CompactionContext> requestCompaction(int priority,
     // Before we do compaction, try to get rid of unneeded files to simplify things.
     removeUnneededFiles();
 
+    if (this.region.getTableDescriptor().isSkipMergingCompaction()) {
+      if (!conf.getBoolean("hbase.store.delete.expired.storefile", true)) {
```

Comment:
Also could you refactor 'hbase.store.delete.expired.storefile' into a variable along with its default value? I see only one usage of this in the code base, apart from testcode.